garde/src/rules/pattern.rs

```diff
@@ -69,6 +69,59 @@ impl<T: Pattern> Pattern for Option<T> {
     }
 }
 
+#[cfg(all(
+    feature = "regex",
+    all(feature = "js-sys", all(target_arch = "wasm32", target_os = "unknown"))
+))]
+#[doc(hidden)]
+pub mod regex_js_sys {
+    pub use ::js_sys::RegExp;
+
+    use super::*;
+
+    impl Matcher for RegExp {
+        fn is_match(&self, haystack: &str) -> bool {
+            self.test(haystack)
+        }
+    }
+
+    impl AsStr for RegExp {
+        fn as_str(&self) -> &str {
+            "[Not supported in JS]"
+        }
+    }
+
+    pub struct RegExpInvoker(&'static str);
+
+    impl RegExpInvoker {
+        pub const fn new(pattern: &'static str) -> Self {
+            Self(pattern)
+        }
+    }
+
+    impl Matcher for RegExpInvoker {
+        fn is_match(&self, haystack: &str) -> bool {
+            RegExp::new(self.0, "u").is_match(haystack)
```

Comment:
constructing a new regex each time doesn't seem great, can this be cached somehow?

Reply:
I tried. The issue here is that the `js-sys` types are inherently not `Sync`, so we can't really store them inside a `static`. It's a pain

Reply:
yeah but wasm is not multithreaded, so it's fine to use a bit of unsafe code to put it in a static

Reply:
specifically `wasm32-unknown-unknown` will forever be single-threaded, threading is enabled via a different target (one of the `wasi` ones, i dont remember which exactly)

Reply:
Added an unsafe `SyncWrapper` type that can't really be constructed outside of the crate without usage of `unsafe` with the explicit disclaimer that either the value must never be passed or shared between threads, or the inner type has to be `Send` and `Sync`